backend/src/main/java/com/javahelp/backend/data/DynamoDBSurveyResponseStore.java

```diff
@@ -0,0 +1,266 @@
+     *
+     * @param userID {@link String} id of author (user) of this surveyResponse
+     * @param surveyResponse {@link SurveyResponse} to create
+     * @return {@link SurveyResponse} that was created
+     */
+    @Override
+    public SurveyResponse create(String userID, SurveyResponse surveyResponse) {
+        surveyStore.cleanTable();
+        surveyStore.create(surveyResponse.getSurvey());
+
+        String id = UUID.randomUUID().toString();
+        surveyResponse.setID(id);
+
+        Map<String, AttributeValue> item = fromSurveyResponse(userID, surveyResponse);
+
+        PutItemRequest request = new PutItemRequest()
+                .withTableName(tableName)
+                .withItem(item);
+
+        getClient().putItem(request);
+
+        return surveyResponse;
+    }
+
+    /**
+     * @param id {@link String} id of the {@link SurveyResponse}
+     * @return {@link SurveyResponse} with the specified id
+     */
+    @Override
+    public SurveyResponse read(String id){
+        Map<String, AttributeValue> key = new HashMap<>();
+        key.put("id", new AttributeValue(id));
+
+        GetItemRequest request = new GetItemRequest()
+                .withTableName(tableName)
+                .withKey(key);
+
+        GetItemResult result = getClient().getItem(request);
+
+        if (result.getItem() == null) {
+            return null;
+        }
+
+        return surveyResponseFromDynamo(result.getItem());
+    }
+
+    /**
+     * Deletes the specified {@link SurveyResponse}
+     *
+     * @param id {@link String} id of the {@link SurveyResponse} to delete
+     */
+    @Override
+    public void delete(String id) {
+        Map<String, AttributeValue> key = new HashMap<>();
+        key.put("id", new AttributeValue(id));
+
+        DeleteItemRequest request = new DeleteItemRequest()
+                .withTableName(tableName)
+                .withKey(key);
+
+        getClient().deleteItem(request);
+    }
+
+    /**
+     * Removes all {@link SurveyResponse}s in database.
+     * ONLY use during preliminary testing!
+     */
+    @Override
+    public void cleanTable() {
+        ScanRequest scanRequest = new ScanRequest()
+                .withTableName(this.tableName);
+        ScanResult result = getClient().scan(scanRequest);
+
+        for (Map<String, AttributeValue> item : result.getItems()) {
+            if (item != null) {
+                delete(item.get("id").getS());
+            }
+        }
+    }
+
+    /**
+     *
+     * @param userID {@link String} id of the user to be queried.
+     * @return {@link List <SurveyResponse>} of the specified user.
+     */
+    @Override
+    public List<SurveyResponse> readByUser(String userID) {
+        String keyConditionExpression = "user_id = :id";
+        Map<String, AttributeValue> expressionAttributeValues = new HashMap<>();
+        expressionAttributeValues.put("id", new AttributeValue().withS(userID));
+
+        ScanRequest scanRequest = new ScanRequest()
```

Comment:
I'm not really sure where `ScanRequest` comes from, but this is guaranteed not to be `null`, right?

Reply:
The null case is handled in the for loop below this line.